core-helm-chart/templates/deployment.yaml

```diff
@@ -48,6 +48,8 @@ spec:
           value: "{{ .Values.EXTERNAL_HOSTNAME }}"
         - name: EXTERNAL_PORT
           value: "{{ .Values.EXTERNAL_PORT }}"
+        - name: KUBECONFIG_DIR
+          value: "{{ .Values.spaceConfigDirectory }}"
```

Comment:
Why do we need to pass this as parameter ?  It's a for local files and each container is limited to its own FS so why can't the  container just creates this at arbitrary place ? 
If you want to pass the SM config through this as well - maybe it would be better to pass that specific config which is indeed accessed by all containers through volume mount and set a variable in the docker file ?

Reply:
That's a good point.  This parameter is passed on to sub scripts called outside of the helm install and could be visible to users.  However, there is no reason the entry.sh script should get this value all the way up from the helm install rather than create some arbitrary temp directory.  External users not going through helm can still pass an explicit value for this parameter. I'll make this change.